src/dynarec/rv64/dynarec_rv64_660f.c

```diff
@@ -2463,9 +2463,14 @@ uintptr_t dynarec64_660F(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int
                 ed = x1;
                 wback = x3;
             }
-            if (rv64_zbs)
-                BEXT_(x4, ed, gd);
-            else {
+            if (rv64_zbs) {
+                if (rex.w) {
+                    BEXT_(x4, ed, gd);
+                } else {
+                    ANDI(x2, gd, 0x1f);
+                    BEXT_(x4, ed, x2);
+                }
+            } else {
                 ANDI(x2, gd, rex.w ? 0x3f : 0xf);
```

Comment:
@ksco I don't know why here it is `ANDI(x2, gd, rex.w ? 0x3f : 0xf);` but not `ANDI(x2, gd, rex.w ? 0x3f : 0x1f);` (difference on the mask). So I did not replace it before.

Reply:
Oh it must be a terrible typo 😄

Reply:
Okay. Let's fix it in this PR.